### ISSUE TITLE
mailsync: [json.exception.type_error.316] invalid UTF-8 byte

### ISSUE THREAD

**Issue by bennlich** [Foundry376/Mailspring#1381]
##### What operating system are you using?

Debian Stretch

##### What version of Mailspring are you using?

1.5.6-4cb1851b

--

I encounter the issue several times a day, but haven't yet determined the steps to reproduce.

Almost every time I open my laptop, the red "Encountered error while syncing..." popup is visible. When I manually click to continue, syncing always succeeds. The next time I am away from my computer for a bit and come back, the error will have returned.

##### Is this an email sync or connection issue?

This is a mailsync issue that showed up after I upgraded to 1.5.6:

```
7606 [2019-02-20 16:16:30.014] [main] [info] Identity created at ... - using ID Schema 1
7606 [2019-02-20 16:16:30.016] [main] [info] ------------- Starting Sync (myemail@gmail.com) ---------------
7606 [2019-02-20 16:16:30.019] [background] [info] Fetching XOAuth2 access token from Gmail for ...
7606 [2019-02-20 16:16:30.019] [metadata] [info] Metadata delta stream starting...
7606 [2019-02-20 16:16:30.019] [metadata] [info] Metadata delta stream closed.
7606 [2019-02-20 16:16:30.043] [metadata] [critical] 
 ***
 *** A JSON exception occurred during program execution: 
 *** [json.exception.type_error.316] invalid UTF-8 byte at index 0: 0xC0
 ***

7606 [2019-02-20 16:16:30.048] [background] [critical] 
 ***
 *** A JSON exception occurred during program execution: 
 *** [json.exception.type_error.316] invalid UTF-8 byte at index 0: 0xC0
 ***

7606 [2019-02-20 16:16:30.059] [metadata] [critical]  *** Stack trace (line numbers are approximate):
 *** (unknown)  clone
 ***
```

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
The error popup is always regarding my gmail account, but this is the only account I have added to mailspring.

Might be related to https://github.com/Foundry376/Mailspring/issues/1093.

**Comment by Robinhuett** [comment]
I've got the same issue on Arch with Mailspring 1.6.0-10f5c95c.
But my account never syncs (freshly installed mailspring). Strange enough on my notebook everything works perfectly fine.

I already reinstalled mailspring and cleared the directory `~/.config/Mailspring`.

```
23014 [2019-03-24 21:52:26.209] [main] [info] Identity created at 1552684612 - using ID Schema 1
23014 [2019-03-24 21:52:26.210] [main] [info] ------------- Starting Sync (mymail@mydomain.xyz) ---------------
23014 [2019-03-24 21:52:26.212] [background] [info] Marking all folders as `busy`
23014 [2019-03-24 21:52:26.212] [background] [info] Syncing folder list...
23014 [2019-03-24 21:52:26.237] [metadata] [critical] 
 ***
 *** A JSON exception occurred during program execution: 
 *** [json.exception.type_error.316] invalid UTF-8 byte at index 1: 0xFE
 ***

23014 [2019-03-24 21:52:26.252] [metadata] [critical]  *** Stack trace (line numbers are approximate):
 *** (unknown)  clone
 ***
```

**Comment by gomera** [comment]
It happens to me as well with `1.6.1` on Linux x86_64

```
2217 [2019-04-26 10:34:26.518] [main] [info] Identity created at 1552762951 - using ID Schema 1
2217 [2019-04-26 10:34:26.519] [main] [info] ------------- Starting Sync (sebastian.mera@gmail.com) ---------------
2217 [2019-04-26 10:34:26.521] [background] [info] Fetching XOAuth2 access token from Gmail for fb0fd5ad
2217 [2019-04-26 10:34:26.521] [metadata] [info] Metadata delta stream starting...
2217 [2019-04-26 10:34:26.522] [metadata] [info] Metadata delta stream closed.
2217 [2019-04-26 10:34:26.538] [metadata] [critical] 
 ***
 *** A JSON exception occurred during program execution: 
 *** [json.exception.type_error.316] invalid UTF-8 byte at index 2: 0x05
 ***

2217 [2019-04-26 10:34:26.542] [background] [critical] 
 ***
 *** A JSON exception occurred during program execution: 
 *** [json.exception.type_error.316] invalid UTF-8 byte at index 0: 0xC0
 ***

2217 [2019-04-26 10:34:26.550] [metadata] [critical]  *** Stack trace (line numbers are approximate):
 *** (unknown)  clone
 ***
```

**Comment by wojciechkepka** [comment]
Same for me with `1.6.1` on Kubuntu 18.04 x86_64
```
2554 [2019-05-15 08:54:28.748] [main] [info] Identity created at 1557897323 - using ID Schema 1
2554 [2019-05-15 08:54:28.749] [main] [info] ------------- Starting Sync (mail@mydomain.com) ---------------
2554 [2019-05-15 08:54:28.752] [background] [info] Marking all folders as `busy`
2554 [2019-05-15 08:54:28.752] [background] [info] Syncing folder list...
2554 [2019-05-15 08:54:28.761] [background] [info] Syncing folder list...
2554 [2019-05-15 08:54:28.765] [background] [info] syncFolderChangesViaCondstore - INBOX: modseq 796 to 796, uidnext 240 to 240
2554 [2019-05-15 08:54:28.766] [background] [info] syncFolderChangesViaCondstore - INBOX/Sent: modseq 69 to 69, uidnext 54 to 54
2554 [2019-05-15 08:54:28.767] [background] [info] syncFolderChangesViaCondstore - INBOX/Drafts: modseq 110 to 110, uidnext 117 to 117
2554 [2019-05-15 08:54:28.769] [background] [info] syncFolderChangesViaCondstore - INBOX/Trash: modseq 116 to 116, uidnext 72 to 72
2554 [2019-05-15 08:54:28.770] [background] [info] syncFolderChangesViaCondstore - INBOX/Mailspring: modseq 10 to 10, uidnext 7 to 7
2554 [2019-05-15 08:54:28.770] [foreground] [info] syncFolderChangesViaCondstore - INBOX: modseq 796 to 796, uidnext 240 to 240
2554 [2019-05-15 08:54:28.770] [background] [info] syncFolderChangesViaCondstore - INBOX/Mailspring/Snoozed: modseq 1 to 1, uidnext 1 to 1
2554 [2019-05-15 08:54:28.771] [background] [info] Sync loop deleting unlinked messages with phase 2.
2554 [2019-05-15 08:54:28.771] [foreground] [info] Idling on folder INBOX
2554 [2019-05-15 08:54:28.772] [background] [info] Sync loop complete.
2554 [2019-05-15 08:54:32.913] [foreground] [info] Idle exited with code 0
2554 [2019-05-15 08:54:32.915] [foreground] [info] syncFolderChangesViaCondstore - INBOX: modseq 796 to 797, uidnext 240 to 241
2554 [2019-05-15 08:54:32.916] [foreground] [info] syncFolderChangesViaCondstore - Changes since HMODSEQ 796: 1 changed, 0 vanished
2554 [2019-05-15 08:54:32.928] [foreground] [info] Idling on folder INBOX
2554 [2019-05-15 08:54:35.802] [main] [info] [local-3a386267-1e82] Running ChangeUnreadTask performLocal:
2554 [2019-05-15 08:54:35.808] [main] [info] [local-3a386267-1e82] -- Succeeded. Changing status to `remote`
2554 [2019-05-15 08:54:36.109] [foreground] [info] Idle exited with code 0
2554 [2019-05-15 08:54:36.111] [foreground] [info] [local-3a386267-1e82] Running ChangeUnreadTask performRemote:
2554 [2019-05-15 08:54:36.117] [foreground] [info] [local-3a386267-1e82] -- Succeeded. Changing status to `complete`
2554 [2019-05-15 08:54:36.119] [foreground] [info] syncFolderChangesViaCondstore - INBOX: modseq 797 to 798, uidnext 241 to 241
2554 [2019-05-15 08:54:36.120] [foreground] [info] syncFolderChangesViaCondstore - Changes since HMODSEQ 797: 1 changed, 0 vanished
2554 [2019-05-15 08:54:36.123] [foreground] [info] Idling on folder INBOX
2554 [2019-05-15 08:54:40.022] [main] [info] [local-60647117-3153] Running ChangeFolderTask performLocal:
2554 [2019-05-15 08:54:40.025] [main] [info] [local-60647117-3153] -- Succeeded. Changing status to `remote`
2554 [2019-05-15 08:54:40.326] [foreground] [info] Idle exited with code 0
2554 [2019-05-15 08:54:40.327] [foreground] [info] [local-60647117-3153] Running ChangeFolderTask performRemote:
2554 [2019-05-15 08:54:40.343] [foreground] [info] [local-60647117-3153] -- Succeeded. Changing status to `complete`
2554 [2019-05-15 08:54:40.345] [foreground] [info] syncFolderChangesViaCondstore - INBOX: modseq 798 to 800, uidnext 241 to 241
2554 [2019-05-15 08:54:40.347] [foreground] [info] syncFolderChangesViaCondstore - Changes since HMODSEQ 798: 0 changed, 1 vanished
2554 [2019-05-15 08:54:40.347] [foreground] [info] - Building queries for range 240-240
2554 [2019-05-15 08:54:40.347] [foreground] [info] Unlinking messages  WHERE remoteFolderId = ? AND remoteUID IN (?) no longer present in remote range.
2554 [2019-05-15 08:54:40.347] [foreground] [info] -- 0 matches.
2554 [2019-05-15 08:54:40.349] [foreground] [info] Idling on folder INBOX
2554 [2019-05-15 08:54:43.753] [metadataExpiration] [info] Scanning for expired metadata
2554 [2019-05-15 08:54:43.753] [metadataExpiration] [info] -- Will wake for next expiration in 7200sec
2554 [2019-05-15 08:54:48.485] [metadata] [critical] 
 ***
 *** A JSON exception occurred during program execution: 
 *** [json.exception.type_error.316] invalid UTF-8 byte at index 0: 0xC0
 ***

2554 [2019-05-15 08:54:48.497] [metadata] [critical]  *** Stack trace (line numbers are approximate):
 *** (unknown)  clone
 ***
```

**Comment by dedenker** [comment]
I have the same error.
Connecting to Office 365

**Comment by zaen323** [comment]
Same here! "Try Again" button works, though seeing a giant red banner every time I open the app is quite annoying.

**Comment by slapula** [comment]
This is happening to me too. I'm on the latest Ubuntu MATE and using gmail for several accounts.  It is particularly annoying if you are trying to send mail and there error pop-up is active.  You end up having to need to clear that error before the message sends.  Not a huge deal but , yeah very annoying.

**Comment by gomera** [comment]
5 months and not even one comment about this, very disappointed Mailspring ...

**Comment by bengotow** [comment]
Hey folks - just a quick heads up that the `invalid UTF-8 byte at index 0` bug will be fixed in the next release this week. This turned out to be a tricky one and the app doesn't retry the connection because the error is actually thrown within the error handling logic for the sync worker. Stay tuned!

**Comment by bengotow** [comment]
Hey folks! This shipped today in 1.7.2 - you can download the update from our website at getmailspring.com/download or choose "Check for Updates" in the app's menu. Enjoy the update!
